adview-manager/src/manager.rs

```diff
@@ -607,4 +604,86 @@ mod test {
         assert_eq!(res.fallback_unit, Some(original_ad_unit));
         assert_eq!(res.campaigns, vec![campaign_0, campaign_1, campaign_2]);
     }
+
+    #[tokio::test]
+    async fn check_if_campaign_is_sticky() {
+        let mut manager = setup_manager("http://localhost:1337".to_string());
+
+        // Case 1 - options has disabled sticky
+        {
+            manager.options.disabled_sticky = true;
+            assert!(!manager.is_campaign_sticky(DUMMY_CAMPAIGN.id).await);
+            manager.options.disabled_sticky = false;
+        }
+        // Case 2 - time is past stickiness treshold, less than 4 minutes ago
+        {
+            let history = vec![HistoryEntry {
+                time: Utc::now() - Duration::days(1), // 24 hours ago
+                unit_id: DUMMY_IPFS[0],
+                campaign_id: DUMMY_CAMPAIGN.id,
+                slot_id: DUMMY_IPFS[1],
+            }];
+            let history = Arc::new(RwLock::new(VecDeque::from(history)));
+
+            manager.history = history;
+
+            assert!(!manager.is_campaign_sticky(DUMMY_CAMPAIGN.id).await);
+        }
+        // Case 3 - time isn't past stickiness treshold, Utc::now()
+        {
+            let history = vec![HistoryEntry {
+                time: Utc::now(),
+                unit_id: DUMMY_IPFS[0],
+                campaign_id: DUMMY_CAMPAIGN.id,
+                slot_id: DUMMY_IPFS[1],
+            }];
+            let history = Arc::new(RwLock::new(VecDeque::from(history)));
+
+            manager.history = history;
+
+            assert!(manager.is_campaign_sticky(DUMMY_CAMPAIGN.id).await);
+        }
+    }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn get_sticky_ad_unit_panic_on_unit_deletion() {
+        let server = MockServer::start().await;
+        let mut manager = setup_manager(server.uri());
+        let history = vec![HistoryEntry {
+            time: Utc::now(),
+            unit_id: DUMMY_AD_UNITS[0].ipfs,
+            campaign_id: DUMMY_CAMPAIGN.id,
+            slot_id: manager.options.market_slot,
+        }];
+        let history = Arc::new(RwLock::new(VecDeque::from(history)));
+        manager.history = history;
+
+        let campaign = Campaign {
+            campaign: DUMMY_CAMPAIGN.clone(),
+            units_with_price: vec![UnitsWithPrice {
+                unit: AdUnit::from(&DUMMY_AD_UNITS[0]),
+                price: UnifiedNum::from_u64(10_000),
+            }],
+        };
+        let res = manager
```

Comment:
I'm not 100% sure if this panics whether the test will pass. This is something we have to be careful with.
This might be panicking instead of the second call.